Understanding WebAssembly better by learning WebAssembly-Text.md

```diff
@@ -209,24 +210,20 @@ Most of the things in this code have been explained. The only new things
 here are `(local $acc i32)`, `block`, `loop`, `br_if $outer` and `$br $loop`.
  - `(local $acc i32)` is like what they call a local variable in higher level
    languages. It is accessed the same way as the parameters, and the first
-   local has the smallest index that is not a parameter.
- - `block $outer` works kind of like `if`, in fact, `if` and `loop` are like
+   local has the smallest assigned index that is not already assigned to a parameter.
+ - `block $outer` works kind of like `if`, in fact, `if` and `loop` are like a
    specialized `block` instruction. The difference is that `block` does not do
    anything special, it just splits the stack and is usually used for general
-   purpouse control flow.
+   purpose control flow.
  - `loop $loop` behaves like a `block`, but if you do a branch on a `loop`,
-   you don't go at the `end`, you go at the top (branches behave like break
-   on `block`s and like continue on `loop`s)
+   you don't go at the `end`, you go to the top (branches behave like `break`
+   on `block`s and like `continue` on `loop`s)
  - `br_if $outer` is a conditional branch/jump instruction. If you know
-   C/C++ you know about jumping using `goto`. But using `goto` you can jump
-   every way, everywhere from every part of the code. WebAssembly is more
-   restrictive. You can branch only out. The labels are there to make 
-   the code easier to read. Without them, it would have been `br_if 1`.
-   As most of the things in WebAssembly, branching is achieved by indexes.
-   Indexing starts from the innermost to the outermost `block`. Which means
-   that the `loop` has an index `0`.
+   C/C++ you know about jumping using `goto`, but by using `goto` you can jump
+   anywhere, from anywhere in the code. WebAssembly is more
```

Comment:
Introduce basic unconditional branching in a separate example first before hopping to a super complex conditional branching example

Reply:
What kind of example will be better suited to simple unconditional branching

Reply:
Some kind of example with a loop